soccer/MainWindow.cpp

```diff
@@ -974,30 +982,44 @@ void MainWindow::channel(int n) {
 // Simulator commands
 
 void MainWindow::on_actionCenterBall_triggered() {
-    SimCommand cmd;
-    cmd.mutable_ball_pos()->set_x(0);
-    cmd.mutable_ball_pos()->set_y(0);
-    cmd.mutable_ball_vel()->set_x(0);
-    cmd.mutable_ball_vel()->set_y(0);
-    _ui.fieldView->sendSimCommand(cmd);
+    grSim_Packet simPacket;
+    grSim_BallReplacement* ball_replace =
+        simPacket.mutable_replacement()->mutable_ball();
+
+    ball_replace->set_x(0);
+    ball_replace->set_y(0);
+    ball_replace->set_vx(0);
+    ball_replace->set_vy(0);
+
+    _ui.fieldView->sendSimCommand(simPacket);
 }
 
 void MainWindow::on_actionStopBall_triggered() {
-    SimCommand cmd;
-    cmd.mutable_ball_vel()->set_x(0);
-    cmd.mutable_ball_vel()->set_y(0);
-    _ui.fieldView->sendSimCommand(cmd);
+    grSim_Packet simPacket;
+    grSim_BallReplacement* ball_replace =
+        simPacket.mutable_replacement()->mutable_ball();
+
+    Geometry2d::Point ballPos =
+        _ui.fieldView->getTeamToWorld() * state()->ball.pos;
+    ball_replace->set_x(ballPos.x());
+    ball_replace->set_y(ballPos.y());
+    ball_replace->set_vx(0);
+    ball_replace->set_vy(0);
+    _ui.fieldView->sendSimCommand(simPacket);
 }
 
 void MainWindow::on_actionResetField_triggered() {
```

Comment:
hmmmm, I like option 1 more, but I'd rather not diverge as much as possible. I think we should do 2 for now, and see if we can't get 1 merged into grSim at some point.